libnavui-speedlimit/src/main/java/com/mapbox/navigation/ui/speedlimit/view/MapboxSpeedInfoView.kt

```diff
@@ -0,0 +1,421 @@
+
+    /**
+     * [AppCompatTextView] that renders MUTCD posted speed.
+     */
+    val speedInfoPostedSpeedMutcd: AppCompatTextView = binding.postedSpeedMutcd
+
+    /**
+     * [AppCompatTextView] that renders MUTCD posted speed unit.
+     */
+    val speedInfoUnitTextMutcd: AppCompatTextView = binding.postedSpeedUnit
+
+    /**
+     * [AppCompatTextView] that renders MUTCD current speed.
+     */
+    val speedInfoCurrentSpeedMutcd: AppCompatTextView = binding.currentSpeedMutcd
+
+    /**
+     * Root VIENNA layout.
+     */
+    val speedInfoViennaLayout: ConstraintLayout = binding.speedInfoViennaLayout
+
+    /**
+     * Root VIENNA layout that renders posted speed child view.
+     */
+    val speedInfoPostedSpeedLayoutVienna: ConstraintLayout = binding.postedSpeedLayoutVienna
+
+    /**
+     * [AppCompatTextView] that renders VIENNA posted speed.
+     */
+    val speedInfoPostedSpeedVienna: AppCompatTextView = binding.postedSpeedVienna
+
+    /**
+     * [AppCompatTextView] that renders VIENNA current speed.
+     */
+    val speedInfoCurrentSpeedVienna: AppCompatTextView = binding.currentSpeedVienna
+
+    init {
+        applyOptions(speedInfoOptions)
+        speedInfoMutcdLayout.setBackgroundResource(R.drawable.background_mutcd_outer_layout)
+        speedInfoPostedSpeedLayoutMutcd.setBackgroundResource(
+            R.drawable.background_mutcd_posted_speed_limit
+        )
+        TextViewCompat.setTextAppearance(
+            speedInfoLegendTextMutcd,
+            R.style.MapboxSpeedInfoMutcdLegendStyle
+        )
+        TextViewCompat.setTextAppearance(
+            speedInfoPostedSpeedMutcd,
+            R.style.MapboxSpeedInfoPostedSpeedMutcdStyle
+        )
+        TextViewCompat.setTextAppearance(
+            speedInfoUnitTextMutcd,
+            R.style.MapboxSpeedInfoMutcdUnitStyle
+        )
+        TextViewCompat.setTextAppearance(
+            speedInfoCurrentSpeedMutcd,
+            R.style.MapboxSpeedInfoCurrentSpeedMutcdStyle
+        )
+
+        speedInfoViennaLayout.setBackgroundResource(R.drawable.background_vienna_outer_layout)
+        speedInfoPostedSpeedLayoutVienna.setBackgroundResource(
+            R.drawable.background_vienna_posted_speed_limit
+        )
+        TextViewCompat.setTextAppearance(
+            speedInfoPostedSpeedVienna,
+            R.style.MapboxSpeedInfoPostedSpeedViennaStyle
+        )
+        TextViewCompat.setTextAppearance(
+            speedInfoCurrentSpeedVienna,
+            R.style.MapboxSpeedInfoCurrentSpeedViennaStyle
+        )
+    }
+
+    /**
+     * Updates this view with posted speed limit and current speed related data.
+     *
+     * @param expected a Expected<SpeedInfoError, SpeedInfoValue>
+     */
+    fun render(expected: Expected<SpeedInfoError, SpeedInfoValue>) {
+        expected.fold(
+            {
+                speedInfoMutcdLayout.isVisible = false
+                speedInfoViennaLayout.isVisible = false
```

Comment:
Can you add an option that allows to display the view even if posted speed limit is not available?
We need this in 1TAP, because users can tap the view and report the missing speed limit for the current road.

Reply:
Done.